### ISSUE TITLE
launchNotification is always returning null

### ISSUE THREAD

**Issue by indrajit-rathod-superk** [aws-amplify/amplify-flutter#3143]
### Description

Amplify is not able to consume the notification when application is in killed state. Amplify receives the notification, when app is in killed state, but on launching the application from notification click, the notification is always returning null including the first call of `Amplify.Notifications.Push.launchNotification`;

### Categories

- [ ] Analytics
- [ ] API (REST)
- [ ] API (GraphQL)
- [ ] Auth
- [ ] Authenticator
- [ ] DataStore
- [ ] Storage

### Steps to Reproduce

1. Trigger a test notification in the pinpoint console
2. Click on a notification, which launches the application
3. Try to consume notification with `final launchNotification = Amplify.Notifications.Push.launchNotification; `
launchNotification value is always returned as null. 

### Platforms

- [ ] iOS
- [X] Android
- [ ] Web
- [ ] macOS
- [ ] Windows
- [ ] Linux

### Flutter Version

3.10.2

### Amplify Flutter Version

1.1.0

### Deployment Method

Custom Pipeline

**Comment by HuiSF** [comment]
Hi @indrajit-rathod-superk thanks for opening this issue. Could you clarify the following point for me to help triage this issue?

1. How did you make the app enter the "killed state" on your Android ?
2. Have you listen to other PN event channels such as background and foreground, have you received anything when you click on the notification to open the app?
3. What's your Android API level?

**Comment by indrajit-rathod-superk** [comment]
Hi @HuiSF, thanks for your response.

   1. I entered the killed state 
       (i). Clicked the menu button, when all apps pop up, swiped the app upward.
       (ii). Forced stoped the app.
       
   2. I have listened to other PN event channels too. Listened to background and foreground. Background event works fine when app is running in background, could receive the notification in notification bar as well as received the push messages as expected. When app is in foreground, I didn’t get notification in notification bar, but received the push message as expected. 
   But, when app is in killed state, I received the notification, but when clicked on notification to open app, received null, nothing else.
   3. Tested on Android API 31 & 33.

**Comment by HuiSF** [comment]
Thanks for the details @indrajit-rathod-superk  I will try to reproduce this issue.

Also a quick follow up question, all the push notifications were sent from Pinpoint, correct?

**Comment by HuiSF** [comment]
Hi @indrajit-rathod-superk  Update from my testing.

Android API 33.
* Kill the App from the App Switcher (swipe up)
  * Send test message from Amazon Pinpoint console notification showed in the notification bar
  * Click the notification to open the App, `Amplify.Notifications.Push.launchNotification` returned corresponding notification payload

* Kill the App by force stopping it (from App Info screen)
  * Send test message from Amazon Pinpoint console notification but no notification was received (this is expected behavior by Android rules)

To sum up, I couldn't reproduce this issue.

`Amplify.Notifications.Push.launchNotification` only returns valid result on the first call after App launched from killed state, and it returns `null` on the subsequent calls. Could you double check if any code calls this API unexpectedly?

**Comment by fjnoyp** [comment]
Hey @indrajit-rathod-superk are you having this issue anymore?  If so could you respond to Hui's response above?

**Comment by fjnoyp** [comment]
Hi we haven't received a comment so we'll close for now.  Feel free to respond or open a new issue if you run into any new problems.
